### ISSUE TITLE
Composer is too narrow / small

### ISSUE THREAD

**Issue by marcoambrosini** [nextcloud/mail#6589]
### Steps to reproduce

1. Compose

### Expected behavior

The composer is reasonably wide

### Actual behavior

The composer is unreasonably narrow 

<img width="1726" alt="Screenshot 2022-05-30 at 16 53 59" src="https://user-images.githubusercontent.com/26852655/171021029-795598c8-af1d-4b99-9388-cd84cd24eeaf.png">

 😅

**Comment by marcoambrosini** [comment]
Leaving a couple of references here: gmail and Protonmail. They're maybe too big but I think that when composing, most of the screen should be occupied by the composer

<img width="1728" alt="Screenshot 2022-05-30 at 17 16 31" src="https://user-images.githubusercontent.com/26852655/171021571-c0c6a45b-9201-487a-878a-902c2e321b7b.png">
<img width="1728" alt="Screenshot 2022-05-30 at 17 17 27" src="https://user-images.githubusercontent.com/26852655/171021666-2f2a6f5e-4a8d-4314-8e5e-b752a08cebc8.png">

**Comment by ChristophWurst** [comment]
@nimishavijay @jancborchardt what width shall we use?

**Comment by szaimen** [comment]
Maybe the large modal size?

**Comment by Bibo-Joshi** [comment]
Another comment in a similar direction is that the width of received mails is rather narrow and doesn't get wider even if you collapse the inbox column to a minimum. See e.g. the dead space around the mail body in the following picture:

![uiae](https://user-images.githubusercontent.com/22366557/174728095-c585d6e0-455a-49ba-922b-981d526ac7a4.png)

I would very much like to have the mail use the full available width (minus some padding) or at least have the possibility to adjust the width like for the inbox column.

I'm commenting this here as a "which fixed width should we use" discussion is apparently ongoing in the linked pr #6717. If I should open a new issue instead, please tell :)

**Comment by jdaviescoates** [comment]
Yes, the ridiculously small pop-up compose window of the current version of Mail is why I cannot event begin to try it out.  It never used to be this bad (as you can still see from the screenshot used on https://apps.nextcloud.com/apps/mail ), but I find this current iteration to be completely unusable.

**Comment by nimishavijay** [comment]
I agree, if it's not possible to interact with the background then it should be bigger. AFAIK the modal composer is a first step towards a popup composer like in Gmail which allows you to browse your mails when you are composing, there it makes sense to have a smaller composer. 
With a modal composer I agree that it can be changed to large size.

**Comment by jancborchardt** [comment]
Let’s look at the default though, not a specific "big" modal composer few people open. Cause the default experience in e.g. Gmail looks like this:

![Gmail composer](https://user-images.githubusercontent.com/925062/187972702-9fa8b4fa-8c13-48e4-a950-4156260bf7db.png)

And this is what we should aim for – not an oversized composer as default.

**Comment by jancborchardt** [comment]
Regarding the main issue here – seems this is because of a font-size issue as it’s not 15px but still 13. https://github.com/nextcloud/mail/pull/6717

**Comment by TheMiNuS** [comment]
It's very hard to edit long and complex email in that small window. An option could be to enable resizing or to detach the window to enable resizing and moving editor to another screen for people with multi screens.

I see this feature request very old now (3Y) and I hope this will be implemented soon as it's one of the reason I switched to another mail client.

**Comment by ChristophWurst** [comment]
Thank you for your feedback

**Comment by sy-subrc** [comment]
Would it be possible or practical to implement a user-specific preference to open the composer dialog maximized by default (similar to the Calendar app)?  

<img width="748" height="873" alt="Image" src="https://github.com/user-attachments/assets/6ba9e4aa-77cb-443f-bc3e-b0729573b0fd" />

similar setting in the Calendar app
<img width="1126" height="439" alt="Image" src="https://github.com/user-attachments/assets/740e9fdd-a2ae-4cdd-9f94-edbc98fd4c9d" />

**Comment by jancborchardt** [comment]
@sy-subrc the last used choice is already remembered, so the effect is already there: If you prefer the wider composer and enable it once, you will always get the wider composer. So no explicit setting needed actually, as it's implicit.

… or at least that's what I thought. :) Currently it remembers only per session, but on refresh this gets forgotten. @ChristophWurst ideally this should just be a per-user remembered implicit preference.

**Comment by jdaviescoates** [comment]
> If you prefer the wider composer

This is the first I've heard that there is a wider composer! 

I admit it's marginally better than the stand insanely small default composer view, but I still can't help wonder why it doesn't just use all the space available? 

It's still using left than half the screen:

<img width="1920" height="959" alt="Image" src="https://github.com/user-attachments/assets/ccfefe37-3a9a-4d48-b55e-cd86a9783c0f" />

Admittedly that's better than the way too narrow standard size:

<img width="1920" height="959" alt="Image" src="https://github.com/user-attachments/assets/f7ebc8d2-0ed4-46b0-9532-fb55b02d508a" />

But why not take a leaf out of Gmails book and use more?

<img width="1920" height="959" alt="Image" src="https://github.com/user-attachments/assets/b8ef4b45-b2bb-4a9c-b2bf-f3502bf7aa8f" />

Roundcube is still better too:

<img width="1920" height="959" alt="Image" src="https://github.com/user-attachments/assets/49393c6e-f267-4fd7-8a3d-6cf4bdd944d8" />

**Comment by sy-subrc** [comment]
@jancborchardt Thank you very much for the feedback. If the last setting were to be remembered permanently, that would be a great solution, in my opinion. And if the expanded Composer dialog were even larger, that would be the icing on the cake ;-)

**Comment by sy-subrc** [comment]
Since most of our users mentioned that the email editor was too small, I modified the CSS in NewMessageModal.vue on our system as follows.
Original CSS:
```
:deep(.modal-wrapper .modal-container) {
	overflow-y: auto !important;
	overflow-x: auto !important;
	// from original Modal max-height
	height: 90%;
	// Max editor + modal height
	max-height: 700px !important;
}
```

Modified CSS:
```
:deep(.modal-wrapper .modal-container) {
    overflow-y: auto !important;
    overflow-x: auto !important;
    // from original Modal max-height
    height: 90%;
    // Max editor + modal height
    max-height: 1500px !important;
    //same for width
    width: 90%;
    max-width: 1500px !important;
}
```
The first impression is very, very good. Now it’s much easier to write longer email messages, and constant scrolling is no longer necessary.
Original:
<img width="2548" height="1346" alt="Image" src="https://github.com/user-attachments/assets/27edde41-79f4-4fc2-8ad9-33eb4bd4ecba" />
Modified:
<img width="2543" height="1255" alt="Image" src="https://github.com/user-attachments/assets/591ba8e6-ecbb-47a6-8701-e252fef4ee8a" />

**Comment by jdaviescoates** [comment]
> The first impression is very, very good. Now it’s much easier to write longer email messages, and constant scrolling is no longer necessary.

I cannot fathom why something like this is not shipped as standard. This issue is 4 years old already.

**Comment by CreatorOfBiontium** [comment]
One of the main reasons we don't fully use Nextcloud Mail is this, it's been open for 4 years and still nothing, I think it needs more attention. Do you think you'll be able to smth with this @ChristophWurst?

**Comment by Circuitlibre** [comment]
@sy-subrc thank you for sharing this CSS modification ! I agree it's very good and much more usable.

If anyone wants to replicate this using [Custom CSS](https://apps.nextcloud.com/apps/theming_customcss), this works reliably :  

```
.modal-wrapper.modal-wrapper--large .modal-container.modal-container {
    overflow-y: auto !important;
    overflow-x: auto !important;
    height: 90vh !important;
    max-height: 90vh !important;
},

.modal-wrapper.modal-wrapper--large .modal-container {
    width: 90vw !important;
    height: 95vh !important;
    max-width: none !important;
    max-height: 850px !important;
}
```

<img width="1864" height="992" alt="Image" src="https://github.com/user-attachments/assets/c11efa38-4c26-44e3-a174-e97f127f832f" />

Note that this also modifies the Settings window : 

<img width="1863" height="993" alt="Image" src="https://github.com/user-attachments/assets/810d2551-bdb2-4480-acd3-171a8a4f45dc" />

**Comment by GretaD** [comment]
hi @jancborchardt @marcoambrosini @nimishavijay, the composer size is brought up again. Should we maybe reconsider it and see if we can add another toggle to have another bigger size?

**Comment by nimishavijay** [comment]
@GretaD I agree, we can have 2 sizes, one the normal small one, and one that is larger with `width: 75vw;` and `height: 90vh`. The current large size is not sufficient as a "large" setting.

**Comment by szimszon** [comment]
Hi,

I don't get it. Why not just make it resize-able?  🤷

**Comment by sy-subrc** [comment]
In my opinion, if the window size is adjustable, it would be important to remember the last window size across sessions. Otherwise, every time you compose a new email, you'd have to set the desired window size all over again.
